internal/future/serving/server.go

```diff
@@ -67,6 +74,22 @@ func NewParamsFromConfig(cfg config.View, prefix string) (*Params, error) {
 	}
 	p := NewParamsFromListeners(grpcLh, httpLh)
 
+	certFile := cfg.GetString("api.tls.certificatefile")
+	privateKeyFile := cfg.GetString("api.tls.privatekey")
+	if len(certFile) > 0 && len(privateKeyFile) > 0 {
+		serverLogger.Printf("Loading TLS certificate from, %s", certFile)
+		publicCertData, err := ioutil.ReadFile(certFile)
+		if err != nil {
+			p.invalidate()
+			return nil, errors.WithStack(err)
+		}
+		privateKeyData, err := ioutil.ReadFile(privateKeyFile)
+		if err != nil {
+			p.invalidate()
+			return nil, errors.WithStack(err)
+		}
+		p.SetTLSConfiguration(publicCertData, publicCertData, privateKeyData)
+	}
 	return p, nil
```

Comment:
Why return the params object again given that the method was called on the object and the object was already modified?

Reply:
This is a standalone function not a method on Params.